test/EFCore.MySql.Tests/Migrations/ModelSnapshotMySqlTest.cs

```diff
@@ -159,11 +159,24 @@ protected void Test(Action<ModelBuilder> buildModel, string expectedCode, Action
             CreateModelValidator().Validate(modelBuilder.Model);
             var model = modelBuilder.Model;
 
-            var generator = new CSharpSnapshotGenerator(new CSharpSnapshotGeneratorDependencies(new CSharpHelper()));
-
-            var builder = new IndentedStringBuilder();
-            generator.Generate("builder", model, builder);
-            var code = builder.ToString();
+            var codeHelper = new CSharpHelper(
+                new MySqlTypeMappingSource(
+                        TestServiceFactory.Instance.Create<TypeMappingSourceDependencies>(),
+                        TestServiceFactory.Instance.Create<RelationalTypeMappingSourceDependencies>(),
+                        new MySqlOptions()));
+
+            var generator = new CSharpMigrationsGenerator(
+                new MigrationsCodeGeneratorDependencies(),
+                new CSharpMigrationsGeneratorDependencies(
+                    codeHelper,
+                    new CSharpMigrationOperationGenerator(
+                        new CSharpMigrationOperationGeneratorDependencies(
+                            codeHelper)),
+                    new CSharpSnapshotGenerator(
+                        new CSharpSnapshotGeneratorDependencies(
+                            codeHelper))));
+
+            var code = generator.GenerateSnapshot("RootNamespace", typeof(DbContext), "Snapshot", model);
```

Comment:
It seems like you've started using the CSharpMigrationsGenerator but did not take the tests into consideration, it is now comparing just the builder code to an 'entire file of code'.

Testing:
```csharp
// <auto-generated />
 using System;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.EntityFrameworkCore.Infrastructure;
 using Microsoft.EntityFrameworkCore.Storage.ValueConversion;

 namespace RootNamespace
 {
     [DbContext(typeof(DbContext))]
     partial class Snapshot : ModelSnapshot
     {
         protected override void BuildModel(ModelBuilder modelBuilder)
         {
 #pragma warning disable 612, 618
             modelBuilder
                 .HasAnnotation("ProductVersion", "2.2.0-rtm-35687")
                 .HasAnnotation("Relational:MaxIdentifierLength", 64);

             modelBuilder.Entity("Pomelo.EntityFrameworkCore.MySql.Migrations.ModelSnapshotMySqlTest+DataTypesVariable", b =>
                 {
                     b.Property<int>("Id")
                         .ValueGeneratedOnAdd();

                     b.Property<byte[]>("TypeByteArray")
                         .IsRequired();

                     b.Property<byte[]>("TypeByteArray255")
                         .IsRequired()
                         .HasMaxLength(255);

                     b.Property<byte[]>("TypeByteArray255N")
                         .HasMaxLength(255);

                     b.Property<byte[]>("TypeByteArrayN");

                     b.Property<string>("TypeJsonArray")
                         .IsRequired()
                         .HasColumnType("json");

                     b.Property<string>("TypeJsonArrayN")
                         .HasColumnType("json");

                     b.Property<string>("TypeJsonObject")
                         .IsRequired()
                         .HasColumnType("json");

                     b.Property<string>("TypeJsonObjectN")
                         .HasColumnType("json");

                     b.Property<string>("TypeString")
                         .IsRequired();

                     b.Property<string>("TypeString255")
                         .IsRequired()
                         .HasMaxLength(255);

                     b.Property<string>("TypeString255N")
                         .HasMaxLength(255);

                     b.Property<string>("TypeStringN");

                     b.HasKey("Id");

                     b.ToTable("DataTypesVariable");
                 });
 #pragma warning restore 612, 618
         }
     }
 }
```

vs the old
```csharp
builder
     .HasAnnotation("Relational:MaxIdentifierLength", 64);

 builder.Entity("Pomelo.EntityFrameworkCore.MySql.Migrations.ModelSnapshotMySqlTest+DataTypesVariable", b =>
     {
         b.Property<int>("Id")
             .ValueGeneratedOnAdd();

         b.Property<byte[]>("TypeByteArray")
             .IsRequired();

         b.Property<byte[]>("TypeByteArray255")
             .IsRequired()
             .HasMaxLength(255);

         b.Property<byte[]>("TypeByteArray255N")
             .HasMaxLength(255);

         b.Property<byte[]>("TypeByteArrayN");

         b.Property<string>("TypeJsonArray")
             .IsRequired()
             .HasColumnType("json");

         b.Property<string>("TypeJsonArrayN")
             .HasColumnType("json");

         b.Property<string>("TypeJsonObject")
             .IsRequired()
             .HasColumnType("json");

         b.Property<string>("TypeJsonObjectN")
             .HasColumnType("json");

         b.Property<string>("TypeString")
             .IsRequired();

         b.Property<string>("TypeString255")
             .IsRequired()
             .HasMaxLength(255);

         b.Property<string>("TypeString255N")
             .HasMaxLength(255);

         b.Property<string>("TypeStringN");

         b.HasKey("Id");

         b.ToTable("DataTypesVariable");
     });
```